src/DataTypes/Serializations/SerializationBool.cpp

```diff
@@ -0,0 +1,138 @@
+#include <DataTypes/Serializations/SerializationBool.h>
+
+#include <Columns/ColumnsNumber.h>
+#include <Common/Exception.h>
+#include <IO/WriteBuffer.h>
+#include <IO/ReadBuffer.h>
+#include <IO/ReadHelpers.h>
+#include <IO/WriteHelpers.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int CANNOT_PARSE_DOMAIN_VALUE_FROM_STRING;
+    extern const int ILLEGAL_COLUMN;
+}
+
+SerializationBool::SerializationBool(const SerializationPtr &nested_)
+        : SerializationCustomSimpleText(nested_)
+{
+}
+
+void SerializationBool::serializeText(const IColumn &column, size_t row_num, WriteBuffer &ostr, const FormatSettings &) const
+{
+    const auto *col = checkAndGetColumn<ColumnUInt8>(&column);
+    if (!col)
+        throw Exception("Bool type can only serialize columns of type UInt8." + column.getName(),
+                        ErrorCodes::ILLEGAL_COLUMN);
+
+    if (col->getData()[row_num])
+        ostr.write(str_true, sizeof(str_true) - 1);
+    else
+        ostr.write(str_false, sizeof(str_false) - 1);
+}
+
+void SerializationBool::deserializeText(IColumn &column, ReadBuffer &istr, const FormatSettings &) const
+{
+    ColumnUInt8 *col = typeid_cast<ColumnUInt8 *>(&column);
+    if (!col)
+    {
+        throw Exception("Bool type can only deserialize columns of type UInt8." + column.getName(),
+                        ErrorCodes::ILLEGAL_COLUMN);
+    }
+
+    if (!istr.eof())
+    {
+        bool value = false;
+
+        if (*istr.position() == 't' || *istr.position() == 'f')
```

Comment:
This looks inconsistent with `deserializeTextEscaped`

Reply:
My original idea was that serializeText/deserializeText only supports some standard bool expressions, while serializeTextEscaped/deserializeTextEscaped can support a variety of custom bool expressions.
But I don’t know what needs we will actually encounter, so if we want to keep the behavior of these two methods consistent, I think it’s okay.

Reply:
I think we should support different representations only for TSV/CSV formats.

Reply:
And Values.
All text formats except JSON.

Reply:
But for VALUES format we should also support TRUE/FALSE.

Reply:
> 
ok, i will update it